connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufSerializer.scala

```diff
@@ -52,6 +52,10 @@ private[sql] class ProtobufSerializer(
         rootCatalystType match {
           case st: StructType =>
             newStructConverter(st, rootDescriptor, Nil, Nil).asInstanceOf[Any => Any]
+          case _ =>
+            throw QueryCompilationErrors.cannotConvertSqlTypeToProtobufError(
+              rootDescriptor.getName,
+              rootCatalystType)
```

Comment:
We're already catching this below. Can we update that instead, and describe when this can happen?

Reply:
@HyukjinKwon the catching below is only for `AnalysisException`, we are not catching MatchError here.

Without this change, the new unit test will fail.

Reply:
Yeah but this isn't the public API. Does it happen with `to_protobuf`? If so, we should fix that `ToProtobuf.checkInputDataTypes` expression to correctly throw an analysis exception.

Reply:
If this can't happen with the public API, simply adding an assert should be good enough.

Reply:
It can be seen in public API.